staging/kube-oidc-proxy/templates/deployment.yaml

```diff
@@ -51,6 +51,8 @@ spec:
           - "--oidc-username-claim=$(OIDC_USERNAME_CLAIM)"
           {{- if or .Values.oidc.caPEM .Values.oidc.caSecretName }}
           - "--oidc-ca-file=/etc/oidc/oidc-ca.pem"
+          {{- else if and .Values.oidc.caCertPemHostPath (not .Values.oidc.caPEM) (not .Values.oidc.caSecretName) }}
+          # disabled --oidc-ca-file to use hostVolume mount
```

Comment:
See my comments below about hostpath. This would be my recommendation:

1. We introduce another knob in `values.yaml` like the following
```yaml
oidc:
  caPEM:
  caSecretName:
  caKubernetesDefault: true
```

At most one of them should be set (or set to true). If all of them are not set (and set to false), `--oidc-ca-file` flag won't be set in the deployment.